src/tilemaker.cpp

```diff
@@ -281,24 +290,43 @@ int main(int argc, char* argv[]) {
 			attributeStore.sortOsmAttributes();
 		}
 
+		// Launch the pool with threadNum threads
+		boost::asio::thread_pool pool(threadNum);
+
+		// Mutex is hold when IO is performed
+		std::mutex io_mutex;
+
 		for (uint zoom=sharedData.config.startZoom; zoom<=sharedData.config.endZoom; zoom++) {
 
-			tileData.SetZoom(zoom);
-			sharedData.zoom = zoom;
+			// Loop through tiles
+			uint tc = 0;
```

Comment:
We should move this outside the loop to be consistent with `total_tiles`, I think?

Reply:
Ah yes, I noticed the counter was wrong, but that is why

Reply:
Zoom level 14, writing tile 38000 of 51383

Reply:
And I added a print when the final tile is generated.